### PR TITLE
Rake tasks to insert discussion topics and migrate pending staff reply.

### PR DESCRIPTION
**Pull request by fonglh**
In preparation for re-parenting of comments to Discussion::Topic instead
of Answer.

Prepare the ground for #2179.

References #1967.

[ci skip]

### CODE REVIEW

**Comment by weiqingtoh** [comment]
Just to check, `pending_staff_reply` must still be checked during reparenting right?

**Comment by fonglh** [comment]
Yes, because there might have been new comments on `Answer` which have reset their `pending_staff_reply` flag.

They're only about a few hundred of them so it's relatively cheap.

File: lib/tasks/db/migrate_pending_staff_reply.rake [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      AND pending_staff_reply=true
+  SQL
+end
+
+# Set pending staff reply based on the old topics
+def migrate_pending_staff_reply
+  old_topic_ids = get_pending_staff_reply
+  old_topic_ids = old_topic_ids.map { |x| x['actable_id'] }
+
+  old_topic_ids.each do |old_topic_id|
```
Comment (weiqingtoh) [inline lib/tasks/db/migrate_pending_staff_reply.rake]: I think you can actually:
 1. grab the `actable_id` and `question_id` for the `old_topic_ids` query. 
 2. grab the ids of all the relevant `Course::Assessment::SubmissionQuestion`
 3. grab all the objects with the `actable_ids` in 2, and `actable_type = Course::Assessment::SubmissionQuestion` and update_column to `true`

Comment (weiqingtoh) [inline lib/tasks/db/migrate_pending_staff_reply.rake]: If it's a few hundred then either way is fine, I think not much of a difference.

**Comment by allenwq** [comment]
@fonglh Instead of using `NOW()`, Can we keep the timestamps of the old discussion topic?  It could be used when ordering the topics.

File: lib/tasks/db/migrate_pending_staff_reply.rake (hunk 1 above)
Comment (fonglh) [inline lib/tasks/db/migrate_pending_staff_reply.rake]: Step 2 is actually the same thing as re-parenting, which is quite complex.

**Comment by fonglh** [comment]
Which one? There might be multiple old discussion topics which map to a single new discussion topic.

It's also quite another 2 joins or so. The existing `SELECT` query is already quite slow.

**Comment by allenwq** [comment]
Any one of them, better to use the last one.  Think we have to do it even it's complex >< otherwise the order of topics will go random.

**Pull request merged**